pkg/partials/interface.go

```diff
@@ -0,0 +1,11 @@
+package partials
+
+// PartialPayload is an interface that is implemented by each of the partial
+// payload types that allows for error messages to be created using the provided
+// error and request payload, that matches the correct format required by the
+// request type. As well as for accounting the weight of the request payload,
+// which is determined by the request's method field.
+type PartialPayload interface {
```

Comment:
Not the biggest fan of `PartialPayload`. Why don't we just call it `RPCPayload`?

I think our goal is to the payload type transparent (i.e. bytes, proxyable, etc) to us, but it's not partial. Might need to update the comment too.

Reply:
I think it is partial, it contains only some of the parts of the request - its parts-ial lmao. But I can change it if you think thatll make it clearer